### ISSUE TITLE
SyntaxError with newer scikit-learn versions

### ISSUE THREAD

**Issue by kiefernforst** [abarker/strip-hints#22]
Dear Mr. Barker,

thanks for your great tool, I'm maintaining hundreds of _type-stripped_ Python 3.10 python packages with it.

scikit-learn however fails due to a parsing error of https://github.com/scikit-learn/scikit-learn/blob/0eb76eb30587d5baa99f43222ce10493d32f13dd/sklearn/linear_model/_logistic.py#L2445

>     coef_init = xp.mean(coefs_paths[0, :, *best_index_int, :], axis=0)

The old syntax has been `coef_init = np.mean(coefs_paths[:, :, best_index, :], axis=1)`, replaced by https://github.com/scikit-learn/scikit-learn/commit/6a9ce612bcc0a75bfdeca5a4f6fc626e88b89d9b
